From khali@linux-fr.org Mon Jun 12 12:41:03 2006
Date: Mon, 12 Jun 2006 21:40:53 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: LM Sensors <lm-sensors@lm-sensors.org>, Peter Korsgaard <jacmet@sunsite.dk>
Subject: [PATCH 01/14] i2c-ocores: Minor cleanups
Message-Id: <20060612214053.4c7ebc0c.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-opencores-cleanup.patch

From: Peter Korsgaard <jacmet@sunsite.dk>

Minor cleanup to the i2c-ocores driver.

Peter Korsgaard will maintain the i2c-ocores driver.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 MAINTAINERS                     |    6 ++++++
 drivers/i2c/busses/i2c-ocores.c |    6 ++----
 2 files changed, 8 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/MAINTAINERS
+++ gregkh-2.6/MAINTAINERS
@@ -2057,6 +2057,12 @@ M:      adaplas@pol.net
 L:      linux-fbdev-devel@lists.sourceforge.net
 S:      Maintained
 
+OPENCORES I2C BUS DRIVER
+P:	Peter Korsgaard
+M:	jacmet@sunsite.dk
+L:	lm-sensors@lm-sensors.org
+S:	Maintained
+
 ORACLE CLUSTER FILESYSTEM 2 (OCFS2)
 P:	Mark Fasheh
 M:	mark.fasheh@oracle.com
--- gregkh-2.6.orig/drivers/i2c/busses/i2c-ocores.c
+++ gregkh-2.6/drivers/i2c/busses/i2c-ocores.c
@@ -38,8 +38,8 @@ struct ocores_i2c {
 #define OCI2C_PREHIGH		1
 #define OCI2C_CONTROL		2
 #define OCI2C_DATA		3
-#define OCI2C_CMD		4
-#define OCI2C_STATUS		4
+#define OCI2C_CMD		4 /* write only */
+#define OCI2C_STATUS		4 /* read only, same address as OCI2C_CMD */
 
 #define OCI2C_CTRL_IEN		0x40
 #define OCI2C_CTRL_EN		0x80
@@ -209,8 +209,6 @@ static struct i2c_adapter ocores_adapter
 	.name		= "i2c-ocores",
 	.class		= I2C_CLASS_HWMON,
 	.algo		= &ocores_algorithm,
-	.timeout	= 2,
-	.retries	= 1,
 };
 
 
